### PR TITLE
WIP: Add strip-ext and change split-ext and strip-ext to return full paths

### PR DESCRIPTION
**Pull request by corasaurus-hex**
```clojure
(fs/strip-ext "/path/to/file-name.html.template" "html.template")
;; => "/path/to/file-name"
```

### CODE REVIEW

File: src/babashka/fs.cljc [hunk 1]
```diff
@@ -670,14 +674,33 @@
...
+        [new-path ext])
+      [path nil])))
+
+(defn strip-ext
+  "Returns the path with the extension removed. If provided, a specific extension will be removed."
+  ([path]
+   (-> path split-ext first))
+  ([path extension]
+   (let [name (file-name path)
+         extension (if (str/starts-with? extension ".")
```
Comment (borkdude) [inline src/babashka/fs.cljc:694]: Let's just make a convention that the function must be called with the extension without a dot, so we don't have to be clever here.

File: src/babashka/fs.cljc [hunk 2]
```diff
@@ -49,6 +49,10 @@
   ([parent child & more]
    (reduce path (path parent child) more)))
 
+(def ^:private path*
```
Comment (borkdude) [inline src/babashka/fs.cljc:52]: Why is this needed?

File: src/babashka/fs.cljc [hunk 3]
```diff
@@ -670,14 +674,33 @@
...
+      (let [new-name (subs name 0 i)
+            new-path (-> (parent path) (path* new-name) str)]
+        [new-path ext])
+      [path nil])))
+
+(defn strip-ext
+  "Returns the path with the extension removed. If provided, a specific extension will be removed."
+  ([path]
+   (-> path split-ext first))
+  ([path extension]
```
Comment (borkdude) [inline src/babashka/fs.cljc:692]: I do think a map as the second arg is more consistent and prevents us programming ourselves into a corner. Apart from `:extension` we could also support a `:regex` in the future for example.

File: src/babashka/fs.cljc (hunk 2 above)
Comment (corasaurus-hex) [reply src/babashka/fs.cljc:52]: So we can use `path` as an argument to functions and still call the original `path` function. Is there another way to accomplish this?

```clojure
(defn foo [arg]
  (println "hi" arg))

(def ^:private foo* foo)

(defn bar [foo]
  (foo* foo))
```

File: src/babashka/fs.cljc (hunk 3 above)
Comment (corasaurus-hex) [reply src/babashka/fs.cljc:692]: ahh, good call. it would also allow `:suffix` later if that's something we wanted to do
